### DISCUSSION TITLE
render RefinableString106 containing string JSON array into HTML table using Handlebars template

### DISCUSSION THREAD

**Discussion by stormanh** [microsoft-search/pnp-modern-search#1921] [Q&A] [answered]
Hello.

Is it possible to render a RefinableString106 that holds a string JSON array into a HTML table using Handlebar expressions?

eg. RefinableString106 = `[{"InvoiceNumber":"20-13139","InvoiceDate":"10-13-2020","InvoiceAmount":"$949.20"},{"InvoiceNumber":"20-13140","InvoiceDate":"10-13-2020","InvoiceAmount":"$734.50"},
{"InvoiceNumber":"20-13141","InvoiceDate":"10-13-2020","InvoiceAmount":"$1,864.50"},
{"InvoiceNumber":"20-13146","InvoiceDate":"10-13-2020","InvoiceAmount":"$226.00"},
{"InvoiceNumber":"20-13153","InvoiceDate":"10-13-2020","InvoiceAmount":"$1,228.88"},
{"InvoiceNumber":"20-13199","Date":"10-20-2020","InvoiceAmount":"$2,853.25"}]`

Have the above rendered as a HTML table so that the search results can be easily read for this managed property?
`<table>
   <thead>
       <tr>
           <th>Invoice Number</th>
           <th>Invoice Date</th>
           <th>Invoice Amount</th>
       </tr>
   </thead>
   <tbody>       <tr>
           <td>20-13139</td>
           <td>10-13-2020</td>
           <td>$949.20</td>
       </tr>       <tr>
           <td>20-13140</td>
           <td>10-13-2020</td>
           <td>$734.50</td>
       </tr>       <tr>
           <td>20-13141</td>
           <td>10-13-2020</td>
           <td>$1,864.50</td>
       </tr>       <tr>
           <td>20-13146</td>
           <td>10-13-2020</td>
           <td>$226.00</td>
       </tr>       <tr>
           <td>20-13153</td>
           <td>10-13-2020</td>
           <td>$1,228.88</td>
       </tr>       <tr>
           <td>20-13199</td>
           <td>10-20-2020</td>
           <td>$2,853.25</td>
       </tr>   </tbody>
</table>`

If so, can someone provide what the handlebar code would be? Thank you!

**Comment by wobba** [comment, marked as answer]
You can start with https://github.com/helpers/handlebars-helpers#jsonparse, or create a custom web component.

**Reply by stormanh** [reply]
Hi @wobba.

I had tried JSONparse without success previously, but after doing some more investigation got it to work using:

      {{#each (JSONparse RefinableString106)}}
       <tr>
           <td>{{InvoiceNumber}}</td>
           <td>{{Date}}</td>
           <td>{{InvoiceAmount}}</td>
       </tr>
       {{/each}}
       
 Thank you!
